File: src/extensions/lang/ruby/qhsm/testsamplehsm1inst.rb
require 'qhsm.rb'
require 'testsamplehsm1.rb'

class TestSample1Impl < TestSample1
    def enterStateX()
        puts "enterStateX"
    end
    def exitStateX()
        puts "exitStateX"
    end
    def enterState0()
        puts "enterState0"
    end
    def exitState0()
        puts "exitState0"
    end
    def enterState1()
        puts "enterState1"
    end
    def exitState1()
        puts "exitState1"
    end
    def sayHello1()
        puts "Hello1"
    end
    def sayHello2()
        puts "Hello2"
    end
    def sayHello3()
        puts "Hello3"
    end
    def Ok(ev)
        return ev.qdata
    end
end


    

def printCurrentStateName(hsm)
    puts "Get currentStateName"
    currentStateName = hsm.currentStateName()
    puts "CurrentStateName:", currentStateName    
end

def test()
    t = TestSample1Impl.new()
    t.init()
    printCurrentStateName(t)
    t.dispatch(qeventFromSignalAndData("Hello", true))
    t.dispatch(qeventFromSignal("Hello"))
    printCurrentStateName(t)
    t.dispatch(qeventFromSignal("Hello"))
    t.dispatch(qeventFromSignalAndData("Hello", true))
    printCurrentStateName(t)
end

test()

File: src/extensions/lang/python/qhsm/testsamplehsm1inst.py
from qhsm import QEventFromSignalAndData, QEventFromSignal
import testsamplehsm1

class TestSample1Impl(testsamplehsm1.TestSample1):
    def enterStateX(self):
        print "enterStateX"
        pass
    def exitStateX(self):
        print "exitStateX"
        pass
    def enterState0(self):
        print "enterState0"
        pass
    def exitState0(self):
        print "exitState0"
        pass
    def enterState1(self):
        print "enterState1"
        pass
    def exitState1(self):
        print "exitState1"
        pass
    def sayHello1(self):
        print "Hello1"
    def sayHello2(self):
        print "Hello2"
    def sayHello3(self):
        print "Hello3"
    def Ok(self, ev):
        return ev.QData
    

def printCurrentStateName(hsm):
    print "Get currentStateName"
    currentStateName = hsm.currentStateName()
    print "CurrentStateName:", currentStateName    

def test():
    t = TestSample1Impl()
    t.init()
    printCurrentStateName(t)
    t.dispatch(QEventFromSignalAndData("Hello", True))
    t.dispatch(QEventFromSignal("Hello"))
    printCurrentStateName(t)
    t.dispatch(QEventFromSignal("Hello"))
    t.dispatch(QEventFromSignalAndData("Hello", True))
    printCurrentStateName(t)


if __name__ == "__main__":
    test()